README.rst

```diff
@@ -26,3 +26,371 @@ Run `tox` to run the whole test suite:
+           super().__init__(board_cls={
+               'native': native.NativeRIOTCtrl,
+           })
+           self.ctrl_list = list()
+
+       def __enter__(self):
+           return self
+
+       def __exit__(self, *exc):
+           for ctrl in self.ctrl_list:
+               ctrl.stop_term()
+
+       def get_ctrl(self, application_directory='.', env=None):
+           # retrieve a RIOTCtrl Object
+           ctrl = super().get_ctrl(
+               env=env,
+               application_directory=application_directory
+           )
+           # append ctrl to list
+           self.ctrl_list.append(ctrl)
+           # flash and start terminal
+           ctrl.flash()
+           ctrl.start_term()
+           # return ctrl with started terminal
+           return ctrl
+
+And the script itself can be re-written as:
+
+.. code:: python
+
+   with RIOTCtrlAppFactory() as factory:
+       env = {'BOARD': 'native'}
+       ctrl = factory.get_ctrl(env=env)
+       shell = SaulShell(ctrl)
+       parser = SaulShellCmdParser()
+       print(parser.parse(shell.saul_cmd()))
+
+GNRC Networking example native
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Lets put all the above into practice and script an experiment verifying
+connectivity between two ctrls, here multiple ``native`` instance will
+be used.
+
+First create two tap interfaces:
+
+.. code:: shell
+
+   sudo ${RIOTBASE}/dist/tools/tapsetup/tapsetup -c 2
+
+Then we can ping and parse the results asserting than packet loss is
+under a threshold or that an mount of responses was received..
+
+.. code:: python
+
+   from riotctrl_shell.gnrc import GNRCICMPv6Echo, GNRCICMPv6EchoParser
+   from riotctrl_shell.netif import Ifconfig
+
+
+   class Shell(ifconfig, GNRCICMPv6Echo):
+     pass
+
+
+   with RIOTCtrlAppFactory() as factory:
+       # Create two native instances, specifying the tap interface
+       native_0 = factory.get_ctrl(env={'BOARD':'native', 'PORT':'tap0'})
+       native_1 = factory.get_ctrl(env={'BOARD':'native', 'PORT':'tap1'})
+       # `NativeRIOTCtrl` allows for `make reset` with `native`
+       native_0.reset()
+       native_1.reset()
+       # Perform a multicast ping and parse results
+       pinger = Shell(native_0)
+       parser = GNRCICMPv6EchoParser()
+       result = parser.parse(pinger.ping6("ff02::1"))
+       assert result['stats']['packet_loss'] < 10    # assert packetloss is under 10%
+       assert result['stats']['rx'] > 0              # assert at least one responder
+
+A more complex example can be seen in the Release Tests:
+`04-single-hop-6lowpan-icmp <https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/test_spec04.py>`__
+
+Examples
+~~~~~~~~
+
+-  pytest: `ReleaseSpecs <https://github.com/RIOT-OS/Release-Specs>`__
+-  unittests:
+   `tests/turo <https://github.com/RIOT-OS/RIOT/blob/master/tests/turo/tests/01-run.py>`__,
+   `tests/congure_test <https://github.com/RIOT-OS/RIOT/blob/master/tests/congure_test/tests/01-run.py>`__
```

Comment:
Didn't we agree during the breakout session to also add the discussion section in one way or another? Something like "Future extensions": CoAP/CBOR, Security checks, ...

Reply:
Added part of the discussion section